tests/test_input.py

```diff
@@ -576,6 +588,83 @@ def test_config_parsing_bad_objective():
     yield check_config_parsing_value_error, config_path
 
 
+def test_config_parsing_bad_objective_2():
+    """
+    Test to ensure config file parsing raises an error with a grid objective given as a list
+    """
+
+    train_dir = join(_my_dir, 'train')
+    test_dir = join(_my_dir, 'test')
+    output_dir = join(_my_dir, 'output')
+
+    # make a simple config file that has a bad task
+    # but everything else is correct
+    values_to_fill_dict = {'experiment_name': 'config_parsing',
+                           'task': 'evaluate',
+                           'train_directory': train_dir,
+                           'test_directory': test_dir,
+                           'featuresets': "[['f1', 'f2', 'f3']]",
+                           'learners': "['LogisticRegression']",
+                           'log': output_dir,
+                           'results': output_dir,
+                           'objective': "['accuracy']"}
+
+    config_template_path = join(_my_dir, 'configs',
+                                'test_config_parsing.template.cfg')
+    config_path = fill_in_config_options(config_template_path,
+                                         values_to_fill_dict,
+                                         'bad_objective')
+
+    yield check_config_parsing_type_error, config_path
+
+
+def test_config_parsing_bad_objectives():
+    """
+    Test to ensure config file parsing raises an error with a grid objectives given as a string
+    """
+
+    train_dir = join(_my_dir, 'train')
+    test_dir = join(_my_dir, 'test')
+    output_dir = join(_my_dir, 'output')
+
+    # make a simple config file that has a bad task
+    # but everything else is correct
+
+    config_template_path = join(_my_dir, 'configs',
+                                'test_objectives_error.template.cfg')
+    values_to_fill_dict = {'train_directory': train_dir,
+                           'log': output_dir,
+                           'results': output_dir}
```

Comment:
I hate to say it but a better solution would be to add `objectives` to `fill_in_config_options` because that's something we would want anyway for future versions.